src/transaction.ts

```diff
@@ -19,90 +19,120 @@ import {promisifyAll} from '@google-cloud/promisify';
 import * as arrify from 'arrify';
 import {EventEmitter} from 'events';
 import {CallOptions} from 'google-gax';
-import {ServiceError} from 'grpc';
 import * as is from 'is';
-import {common as p} from 'protobufjs';
 import {Readable} from 'stream';
 
-import {codec, Json, JSONOptions, Type, Value} from './codec';
-import {PartialResultStream, partialResultStream, ResumeToken, Row} from './partial-result-stream';
+import {SpannerClient as s} from './v1';
```

Comment:
Yeah no you're completely right. This honestly came down to what looks like a silly incompatibility between `s.Type` and proto's `IType`, but deeper down it's because the generated files have a different TypeCode system to what's in use at the moment so keeping certain `s.<x>` annotations seemed like the best way forward for the time being. I _could_ make custom types for those situations though, let me know 👍 

Linked it for you here: [Generated TypeCode](https://github.com/googleapis/nodejs-spanner/blob/master/proto/spanner.d.ts#L4376) vs [TypeCode in use](https://github.com/googleapis/nodejs-spanner/blob/master/src/v1/spanner_client.d.ts#L413)

Apart from that, the generated types don't account for every type currently in use like `s.ExecuteBatchDmlRequest` and `s.ExecuteBatchDmlResponse` (it should though, right?), so I kept those as well